angular/projects/spark-angular/src/lib/components/sprk-tabbed-navigation/sprk-tabbed-navigation.component.ts

```diff
@@ -254,24 +232,73 @@ export class SprkTabbedNavigationComponent implements AfterContentInit {
   /**
    * @ignore
    */
-  retreatTab(tabs, tabpanels, activeClass) {
-    const activeIndex = this.getActiveTabIndex(tabs, activeClass);
+  incrementTab(tabs, tabpanels, activeClass, direction) {
+    var activeIndex = this.getActiveTabIndex(tabs, activeClass);
+
+    let foundNewIndex = false;
 
+    // Start looking for the next available tab
+    while (foundNewIndex === false){
+      // if the next tab would be off the left of the tabstrip
+      if (activeIndex + direction < 0) {
+        // loop to the end and keep looking
+        activeIndex = tabs.length;
+
+      // if the next tab would be off the right of the tabstrip
+      } else if (activeIndex + direction >= tabs.length){
+        // loop back to the beginning and keep looking
+        activeIndex = -1;
+
+      // If the next tab is not disabled
+      } else if (!tabs[activeIndex + direction].hasAttribute('disabled')){
+        // move to the next tab
+        activeIndex += direction;
+        // stop looking for the correct tab
+        foundNewIndex = true;
+
+      } else {
+        // move to the next tab and keep looking
+        activeIndex+= direction;
+      }
+    }
+
+    // deselect all tabs
     this.resetTabs(tabs, tabpanels, activeClass);
+    // select the correct tab
+    this.setActiveTab(tabs[activeIndex], tabpanels[activeIndex], activeClass);
+  }
 
-    if (activeIndex - 1 === -1) {
-      this.setActiveTab(
-        tabs[tabs.length - 1],
-        tabpanels[tabs.length - 1],
-        activeClass
-      );
+  goToEndTab(tabs, tabpanels, activeClass, direction) {
+    var newActiveIndex;
+
+    // if direction is positive, go to the right-most tab
+    if (direction > 0) {
+      newActiveIndex = tabs.length-1;
+
+    // else go to the left-most tab
     } else {
-      this.setActiveTab(
-        tabs[activeIndex - 1],
-        tabpanels[activeIndex - 1],
-        activeClass
-      );
+      newActiveIndex = 0;
     }
+
+    let foundNewIndex = false;
+
+    // step through the tabs until we find one that isn't disabled
+    while (foundNewIndex === false){
+
+      // if this tab is not disabled
+      if (!tabs[newActiveIndex].hasAttribute('disabled')){
```

Comment:
Should be a space here btw the closing paren and opening curly @bobwise or @guern1kn

Reply:
On it.